core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
Concurrency bugs are hard to trigger and I disagree that other Spark usage would have exposed them already. I think this is the main point where we disagree.

Assuming that the user has tracked the problem down, deploying the fix may be the hardest part. Many customers can't simply make changes and push a new version without sending code through QA or some other validation. A runtime config change is much easier.